### PR TITLE
refactor: Encapsulate task runner startup to module

### PR DESCRIPTION
**Pull request by tomi**
## Summary

Encapsulate startup of task runners into separate module. This is to make sure the correct classes are loaded conditionally depending on task runner (process, disconnect analyzer).

Also:
- Rename `TaskRunnerService` to `TaskRunnerWsServer`

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/cli/src/runners/task-runner-module.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+		await this.loadTaskManager();
+		await this.loadTaskRunnerServer();
+
+		if (this.runnerConfig.mode !== 'external') {
+			await this.startInternalTaskRunner();
+		}
+	}
+
+	async stop() {
```
Comment (ivov) [inline packages/cli/src/runners/task-runner-module.ts:39]: Looks like this is unused? Should this have an `@OnShutdown` decorator?

**Review by ivov — DISMISSED** [review]

File: packages/cli/src/runners/task-runner-module.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	private taskRunnerWsServer: TaskRunnerWsServer | undefined;
+
+	private taskManager: LocalTaskManager | undefined;
+
+	private taskRunnerProcess: TaskRunnerProcess | undefined;
+
+	constructor(private readonly runnerConfig: TaskRunnersConfig) {}
+
+	async start() {
+		a.ok(!this.runnerConfig.disabled, 'Task runner is disabled');
```
[the remark this answers is not in the document]
Comment (tomi) [reply packages/cli/src/runners/task-runner-module.ts:29]: Fully agree. I don't know why it was initially set this way. Let's change it in a separate PR

File: packages/cli/src/runners/task-runner-module.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	async stop() {
+		if (this.taskRunnerProcess) {
+			await this.taskRunnerProcess.stop();
+		}
+
+		if (this.taskRunnerHttpServer) {
+			await this.taskRunnerHttpServer.stop();
+		}
```
[the remark this answers is not in the document]
Comment (tomi) [reply packages/cli/src/runners/task-runner-module.ts:46]: Added missing undefined assignments

File: packages/cli/src/runners/task-runner-module.ts (hunk 1 above)
Comment (tomi) [reply packages/cli/src/runners/task-runner-module.ts:39]: It's currently used only for tests. My idea is to make the module responsible for shutting down its individual components, but at the moment it's still the individual components that have the `@OnShutdown` decorator. I'll do that as a followup

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.67.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.67.0)

**Pull request merged**